process/peer/process.go

```diff
@@ -342,6 +344,101 @@ func (vs *validatorStatistics) RootHash() ([]byte, error) {
 	return vs.peerAdapter.RootHash()
 }
 
+func (vs *validatorStatistics) getValidatorDataFromLeaves(
+	leaves map[string][]byte,
+) (map[uint32][]*state.ValidatorInfoData, error) {
+
+	validators := make(map[uint32][]*state.ValidatorInfoData, vs.shardCoordinator.NumberOfShards()+1)
+	for i := uint32(0); i < vs.shardCoordinator.NumberOfShards(); i++ {
+		validators[i] = make([]*state.ValidatorInfoData, 0)
+	}
+	validators[sharding.MetachainShardId] = make([]*state.ValidatorInfoData, 0)
+
+	sortedLeaves := make([][]byte, len(leaves))
+	i := 0
+	for _, pa := range leaves {
+		sortedLeaves[i] = pa
+		i++
+	}
+
+	sort.Slice(sortedLeaves, func(i, j int) bool {
+		return bytes.Compare(sortedLeaves[i], sortedLeaves[j]) < 0
+	})
+
+	for _, pa := range sortedLeaves {
+		peerAccount := &state.PeerAccount{}
+		err := vs.marshalizer.Unmarshal(peerAccount, pa)
+		if err != nil {
+			return nil, err
+		}
+
+		currentShardId := peerAccount.CurrentShardId
+		pk128, err := hex.DecodeString(string(peerAccount.BLSPublicKey))
+		validatorInfoData := &state.ValidatorInfoData{
+			PublicKey:                  pk128,
+			ShardId:                    peerAccount.CurrentShardId,
+			List:                       "",
+			Index:                      0,
+			TempRating:                 peerAccount.TempRating,
+			Rating:                     peerAccount.Rating,
+			RewardAddress:              peerAccount.RewardAddress,
+			LeaderSuccess:              peerAccount.LeaderSuccessRate.NrSuccess,
+			LeaderFailure:              peerAccount.LeaderSuccessRate.NrFailure,
+			ValidatorSuccess:           peerAccount.ValidatorSuccessRate.NrSuccess,
+			ValidatorFailure:           peerAccount.ValidatorSuccessRate.NrFailure,
+			NumSelectedInSuccessBlocks: peerAccount.NumSelectedInSuccessBlocks,
+			AccumulatedFees:            big.NewInt(0).Set(peerAccount.AccumulatedFees),
+		}
+
+		validators[currentShardId] = append(validators[currentShardId], validatorInfoData)
+	}
+
+	return validators, nil
+}
+
+// GetValidatorInfosForRootHash returns all the peer accounts from the trie with the given rootHash
+func (vs *validatorStatistics) GetValidatorInfosForRootHash(rootHash []byte) (map[uint32][]*state.ValidatorInfoData, error) {
+	allLeaves, err := vs.peerAdapter.GetAllLeaves(rootHash)
```

Comment:
maybe add here a stop watch instance as to measure the time needed for fetching all leaves and assemble the vInfo map

Reply:
done